AK/IPv6Address.h

```diff
@@ -120,12 +156,31 @@ class [[gnu::packed]] IPv6Address {
 #endif
     }
 
-    static Optional<IPv6Address> from_string(StringView string)
+    static constexpr Optional<IPv6Address> from_string(StringView string)
     {
         if (string.is_null())
             return {};
 
-        auto const parts = string.split_view(':', SplitBehavior::KeepEmpty);
+        // Manual split_at(':', KeepEmpty) implementation, since the StringView one is not constexpr.
+        Array<StringView, 9> parts_creator;
+        size_t current_part = 0;
+        size_t current_part_start = 0;
+        for (size_t index = 0; index < string.length(); ++index) {
+            auto const character = string[index];
+            if (character == ':') {
+                // terminate current part
+                parts_creator[current_part] = string.substring_view(current_part_start, index - current_part_start);
+                ++current_part;
```

Comment:
Please add an explicit range check here, instead of relying on Array for it
if you `VERIFY`  or  `return {};` is your choice

Reply:
Yeah that should just be a parser failure.